vars/cloudFoundryDeploy.groovy

```
@@ -63,42 +63,54 @@ void call(Map parameters = [:]) {
+                    writeFile file: config.smokeTestScript, text: libraryResource(config.smokeTestScript)
+                }
 
-            config.smokeTest = '--smoke-test $(pwd)/' + config.smokeTestScript
-            sh "chmod +x ${config.smokeTestScript}"
+                config.smokeTest = '--smoke-test $(pwd)/' + config.smokeTestScript
+                sh "chmod +x ${config.smokeTestScript}"
 
-            echo "[${STEP_NAME}] CF native deployment (${config.deployType}) with cfAppName=${config.cloudFoundry.appName}, cfManifest=${config.cloudFoundry.manifest}, smokeTestScript=${config.smokeTestScript}"
+                echo "[${STEP_NAME}] CF native deployment (${config.deployType}) with cfAppName=${config.cloudFoundry.appName}, cfManifest=${config.cloudFoundry.manifest}, smokeTestScript=${config.smokeTestScript}"
 
-            dockerExecute (
-                script: script,
-                dockerImage: config.dockerImage,
-                dockerWorkspace: config.dockerWorkspace,
-                stashContent: config.stashContent,
-                dockerEnvVars: [CF_HOME:"${config.dockerWorkspace}", CF_PLUGIN_HOME:"${config.dockerWorkspace}", STATUS_CODE: "${config.smokeTestStatusCode}"]
-            ) {
-                deployCfNative(config)
+                dockerExecute (
+                    script: script,
+                    dockerImage: config.dockerImage,
+                    dockerWorkspace: config.dockerWorkspace,
+                    stashContent: config.stashContent,
+                    dockerEnvVars: [CF_HOME:"${config.dockerWorkspace}", CF_PLUGIN_HOME:"${config.dockerWorkspace}", STATUS_CODE: "${config.smokeTestStatusCode}"]
+                ) {
+                    deployCfNative(config)
+                }
+            }
+        } catch (err) {
+            deploySuccess = false
+            throw err
+        } finally {
+            if (deploy) {
+                reportToInflux(script, config, deploySuccess, jenkinsUtils)
```

I am in favor of calling `reportToInflux` in a try catch and ignoring the exception if any raised.